pkg/github/check_api.go

```diff
@@ -3,68 +3,73 @@ package github
 import (
 	"context"
 	"fmt"
+	"strings"
 	"time"
 
-	"github.com/google/go-github/v39/github"
+	"github.com/google/go-github/v43/github"
 	"github.com/rs/zerolog/log"
+	"github.com/shurcooL/githubv4"
+	"github.com/terraform-tools/terraform-checker/pkg/terraform"
 )
 
-func (t *CheckEvent) UpdateCheckRun(cr GhCheckRun, status bool, message string) {
-	var conclusion string
-	if status {
-		conclusion = "success"
-	} else {
-		conclusion = "failure"
+func (e *CheckEvent) CreateCheckRun(check terraform.TfCheck) (GhCheckRun, error) {
+	checkRunName := fmt.Sprintf("terraform-check %v %v", check.Name(), check.RelDir())
+	log.Info().Msgf("Create check run %s on repo %s PR %s", checkRunName, e.GetRepo().GetFullName(), e.GetPRUrl())
+
+	externalId := fmt.Sprintf("%s:%s", check.RelDir(), check.Name())
+	cr, _, err := e.GetGhClient().Checks.CreateCheckRun(context.TODO(),
+		e.GetRepo().GetOwner().GetLogin(),
+		e.GetRepo().GetName(),
+		github.CreateCheckRunOptions{
+			Name:       checkRunName,
+			HeadSHA:    e.GetSHA(),
+			Status:     github.String(strings.ToLower(string(githubv4.CheckStatusStateInProgress))),
+			StartedAt:  &github.Timestamp{Time: time.Now()},
+			ExternalID: github.String(externalId),
+		})
+	if err != nil {
+		log.Error().Err(err).Msg("Error creating check run")
+
+		return GhCheckRun{}, err
 	}
+	return GhCheckRun{
+		Name: checkRunName,
+		ID:   *cr.ID,
+	}, nil
+}
+
+func (e *CheckEvent) UpdateCheckRun(cr GhCheckRun, conclusion githubv4.CheckConclusionState, cmdOutput string, check terraform.TfCheck) {
+
+	checkStatus := fmt.Sprintf("**Check Status:**  %s", string(CheckConclusionStateEmoji(conclusion)))
 
 	cro := github.CheckRunOutput{
-		Title:   github.String("Terraform check is a " + conclusion),
-		Summary: &message,
-		Text:    github.String(""),
+		Title:       &cr.Name,
+		Summary:     &checkStatus,
+		Annotations: check.Annotations(),
+	}
+	if cmdOutput != "" {
+		cro.Text = github.String(fmt.Sprintf("```shell\n%s\n```", cmdOutput))
 	}
 
-	fmtAction := github.CheckRunAction{
-		Label:       "fmt",
-		Description: "format the terraform code",
-		Identifier:  "fmt",
+	actions := []*github.CheckRunAction{}
+	if conclusion != githubv4.CheckConclusionStateSuccess {
+		actions = check.FixActions()
 	}
-	actions := []*github.CheckRunAction{&fmtAction}
-	_, _, err := t.GhClient.Checks.UpdateCheckRun(context.TODO(),
-		t.Repo.Owner,
-		t.Repo.Name,
+
+	log.Info().Msgf("Update check run %s on repo %s PR %s", cr.Name, e.GetRepo().GetFullName(), e.GetPRUrl())
```

Comment:
log levels will probably need to be reviewed in the future

Reply:
Yep, not sure which ones should stay as Info, maybe the createCheckRun one is enough in Info